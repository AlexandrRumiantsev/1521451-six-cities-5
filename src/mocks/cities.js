export default [
  {
    id: `1`,
    title: `Amsterdam`,
    link: `Amsterdam`,
    lat: 52.3809453943508,
    lon: 4.931309666406198
  },
  {
    id: `2`,
    title: `Hamburg`,
    link: `Hamburg`,
    lat: 53.5611,
    lon: 9.9933
  },
  {
    id: `3`,
    title: `Dusseldorf`,
    link: `Dusseldorf`
  },
  {
    id: `4`,
    title: `Brussels`,
    link: `Brussels`
  },
  {
    id: `5`,
    title: `Cologne`,
    link: `Cologne`
  },
  {
    id: `6`,
    title: `Paris`,
    link: `Paris`
  },
];
