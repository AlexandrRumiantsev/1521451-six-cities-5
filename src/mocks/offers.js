export default [
  {
    id: 1,
    title: `Beautiful & luxurious studio at great location`,
    gallary: [
      {
        id: 11,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 12,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 13,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 14,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 15,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 16,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: true,
    houseLevel: `Apartment`,
    rating: 4.8,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Apartment`
      },
      {
        field: `Count`,
        value: `3 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 1 adults`
      }
    ],
    prices: [`€380`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
      river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
        `An independent House, strategically located between
      Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 2,
    title: `Wood and stone place`,
    gallary: [
      {
        id: 21,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 22,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 23,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 24,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 25,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 26,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 3.8,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `1 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 1 adults`
      }
    ],
    prices: [`€80`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 3,
    title: `Nice, cozy, warm big bed apartment`,
    gallary: [
      {
        id: 31,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 32,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 33,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 34,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 35,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 36,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: [`€180`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 4,
    title: `Canal View Prinsengracht`,
    gallary: [
      {
        id: 41,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 42,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 43,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 44,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 45,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 46,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: [`€180`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 5,
    title: `Nice, cozy, warm big bed apartment`,
    gallary: [
      {
        id: 51,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 52,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 53,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 54,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 55,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 56,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: [`€180`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  }
];
