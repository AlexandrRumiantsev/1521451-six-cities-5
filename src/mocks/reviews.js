export default [
  {
    id: `001`,
    offers: 1,
    user: {
      id: `1`,
      name: `Max`,
      ava: `/img/avatar-max.jpg`
    },
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      April 2019`,
    rating: 2,
  },
  {
    id: `002`,
    offers: 1,
    user: {
      id: `2`,
      name: `Иван`,
      ava: `/img/avatar-max.jpg`
    },
    text: `Отличный вариант для семейных пар!`,
    rating: 4,
  },
];
