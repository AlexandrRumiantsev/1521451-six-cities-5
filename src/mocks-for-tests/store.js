import configureStore from "redux-mock-store";
import {offers, cities, reviews, user, baseFilter} from "./mocks";

const createStore = configureStore();

const mockStore = createStore({
  Offers: {
    currentCity: cities[`Brussels`],
    baseFilter,
    offers,
    offer: offers[0],
    nearby: offers,
    favorites: offers,
    data: offers,
    getFavoriteDispatch: ()=>{}
  },
  Review: {
    reviews
  },
  User: user,
  offer: offers[0]
});

export {mockStore};
