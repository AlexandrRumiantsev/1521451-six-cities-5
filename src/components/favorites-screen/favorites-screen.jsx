import React, { PureComponent } from 'react';
import HeaderContainer from '../header/header-container';
import { getFavorite } from '../../store/actions/offers/offers';
import { connect } from "react-redux";
import { selectFavorites } from "../../store/selectors/offers/filter-favorites";

class FavoritesScreen extends PureComponent {
  constructor(props) {
    super(props);
    this.props.getFavorite()
  }
  render() {
    const { favorites } = this.props;

    return (
      <div className="page">
        <HeaderContainer />
        <main className="page__main page__main--favorites">
          <div className="page__favorites-container container">
            <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <ul className="favorites__list">
                {
                  Object.keys(selectFavorites(favorites)).map(function (cityName) {
                    if (selectFavorites(favorites)[cityName].length != 0) {
                      return (
                        <li className="favorites__locations-items">
                          <div className="favorites__locations locations locations--current">
                            <div className="locations__item">
                              <a className="locations__item-link" href="#">
                                <span>{cityName}</span>
                              </a>
                            </div>
                          </div>
                          <div className="favorites__places">
                            {selectFavorites(favorites)[cityName].map(function (favorite) {
                              return (<article className="favorites__card place-card">
                                <div className="favorites__image-wrapper place-card__image-wrapper">
                                  <a href={`/offer/${favorite.id}`}>
                                    <img className="place-card__image" src={favorite.preview_image} width="150" height="110" alt={favorite.title} />
                                  </a>
                                </div>
                                <div className="favorites__card-info place-card__info">
                                  <div className="place-card__price-wrapper">
                                    <div className="place-card__price">
                                      <b className="place-card__price-value">&euro;{favorite.price}</b>
                                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                                    </div>
                                    <button className="place-card__bookmark-button place-card__bookmark-button--active button" type="button">
                                      <svg className="place-card__bookmark-icon" width="18" height="19">
                                        <use xlinkHref="#icon-bookmark"></use>
                                      </svg>
                                      <span className="visually-hidden">In bookmarks</span>
                                    </button>
                                  </div>
                                  <div className="place-card__rating rating">
                                    <div className="place-card__stars rating__stars">
                                      <span className='rating__w100'></span>
                                      <span className="visually-hidden">Rating</span>
                                    </div>
                                  </div>
                                  <h2 className="place-card__name">
                                    <a href="#">{favorite.title}</a>
                                  </h2>
                                  <p className="place-card__type">{favorite.type}</p>
                                </div>
                              </article>
                              )
                            })
                            }
                          </div>
                        </li>
                      )
                    }
                  })}
              </ul>
            </section>
          </div>
        </main>
        <footer className="footer container">
          <a className="footer__logo-link" href="main.html">
            <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33" />
          </a>
        </footer>
      </div>
    );
  };
}

const mapStateToProps = (state, props) => {
  return {
    favorites: state.Offers.favorites,
  };
};

const mapDispatchToProps = (dispatch) => ({
  getFavorite() {
    dispatch(getFavorite());
  }
});

export { FavoritesScreen };
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesScreen);
