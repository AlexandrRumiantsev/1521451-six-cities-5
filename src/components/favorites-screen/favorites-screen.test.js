import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {FavoritesScreen} from "./favorites-screen";
import {Users, offers} from "../../mocks-for-tests/mocks";

it(`Favorites is rendered correctly`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <FavoritesScreen user={Users} offers={offers} favorites={offers} />
  );
  expect(tree).toMatchSnapshot();
});
