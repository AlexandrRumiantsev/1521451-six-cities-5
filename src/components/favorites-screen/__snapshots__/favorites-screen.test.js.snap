// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites is rendered correctly 1`] = `
<div
  className="page"
>
  <Memo(Connect(HeaderContainer)) />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/34"
                  >
                    <img
                      alt="Nice, cozy, warm big bed apartment"
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        485
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <Link
                      to="/offer/34"
                    >
                      Nice, cozy, warm big bed apartment
                    </Link>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Brussels
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/3"
                  >
                    <img
                      alt="The house among olive "
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        218
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <Link
                      to="/offer/3"
                    >
                      The house among olive 
                    </Link>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Hamburg
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/11"
                  >
                    <img
                      alt="The house among olive "
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        190
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <Link
                      to="/offer/11"
                    >
                      The house among olive 
                    </Link>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/87"
                  >
                    <img
                      alt="Tile House"
                      className="place-card__image"
                      height="110"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        266
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        className="rating__w100"
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <Link
                      to="/offer/87"
                    >
                      Tile House
                    </Link>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    hotel
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <Footer />
</div>
`;
